jina/types/arrays/memmap.py

```diff
@@ -266,3 +271,77 @@ def physical_size(self) -> int:
         :return: the number of bytes
         """
         return os.stat(self._header_path).st_size + os.stat(self._body_path).st_size
+
+    def get_attributes(self, *fields: str) -> Union[List, List[List]]:
+        """Return all nonempty values of the fields from all docs this array contains
+
+        :param fields: Variable length argument with the name of the fields to extract
+        :return: Returns a list of the values for these fields.
+            When `fields` has multiple values, then it returns a list of list.
+        """
+        index = None
+        fields = list(fields)
+        if 'embedding' in fields and self._embeddings_memmap:
+            embeddings = [doc.get_attributes('embedding') for doc in self]
+            self._embeddings_memmap = np.asarray(embeddings)
+            index = fields.index('embedding')
+            fields.remove('embedding')
+        if fields:
+            contents = [doc.get_attributes(*fields) for doc in self]
+            if len(fields) > 1:
+                contents = list(map(list, zip(*contents)))
+            if index:
+                contents.insert(index, embeddings)
+
+        return contents
+
+    def get_attributes_with_docs(
```

Comment:
is this needed?

Reply:
it's one we get also from the mixin class, `DocumentArray` has this function, do we want to remove it?

Reply:
But I mean, why do we need to rewrite it? I think we can still inherit from `DocArrayGetAttrMixin` but we specialize the `get_attributes`